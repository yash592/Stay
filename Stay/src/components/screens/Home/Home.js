import React, { Component } from "react";
import {
  View,
  Image,
  Text,
  TouchableOpacity,
  ImageBackground
} from "react-native";
import { ReservationCard } from "../../common/ReservationCard";
import { Actions } from "react-native-router-flux";
import { Dimensions } from "react-native";

// Home component to show the list of available screens

const Home = () => {
  const width = Dimensions.get("window").width;
  const height = Dimensions.get("window").height;

  return (
    <View style={styles.container}>
      <View>
        <Image
          source={{ uri: "https://i.imgur.com/JARMkib.png" }}
          style={{ width: width * 0.7, height: height * 0.2 }}
        />
      </View>
      <TouchableOpacity style={styles.tile} onPress={() => Actions.listings()}>
        <ImageBackground
          source={{
            uri:
              "https://images.pexels.com/photos/573552/pexels-photo-573552.jpeg"
          }}
          style={{ width: "100%" }}
          imageStyle={{
            width: "100%",
            height: "100%",
            position: "relative",
            opacity: 0.2
          }}
        >
          <View style={styles.textBox}>
            <Text style={styles.text}>RESERVATIONS</Text>
          </View>
        </ImageBackground>
      </TouchableOpacity>

      <TouchableOpacity
        onPress={() => Actions.createlisting()}
        style={styles.tile}
      >
        <ImageBackground
          source={{
            uri:
              "https://images.pexels.com/photos/573552/pexels-photo-573552.jpeg"
          }}
          style={{ width: "100%" }}
          imageStyle={{
            width: "100%",
            height: "100%",
            position: "relative",
            opacity: 0.2
          }}
        >
          <View style={styles.textBox}>
            <Text style={styles.text}>MAKE A RESERVATION</Text>
          </View>
        </ImageBackground>
      </TouchableOpacity>
    </View>
  );
};

const styles = {
  container: {
    flex: 1,
    flexDirection: "column",
    flexWrap: "wrap",
    justifyContent: "center",
    alignItems: "center"
  },
  tile: {
    margin: 15,
    width: "95%",
    height: "10%",
    shadowColor: "black",
    shadowOffset: { width: 10, height: 10 },
    shadowOpacity: 1,
    shadowRadius: 2,
    elevation: 1,
    backgroundColor: "white",
    borderRadius: 2,
    justifyContent: "center",
    alignItems: "center"
  },
  textBox: {
    position: "absolute",
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    padding: 15,
    justifyContent: "flex-start"
  },
  text: {
    fontSize: 22,
    fontWeight: "bold",
    shadowColor: "white",
    textShadowOffset: { width: 0.1, height: 0.1 },
    shadowOpacity: 0.5,
    textShadowRadius: 1,
    fontFamily: "sans-serif-condensed"
  }
};

export { Home };
