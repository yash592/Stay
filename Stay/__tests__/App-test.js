/**
 * @format
 */

import "react-native";
import renderer from "react-test-renderer";
import React from "react";
import { Home } from "../src/components/screens/Home";

// Note: test renderer must be required after react-native.
// import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});
