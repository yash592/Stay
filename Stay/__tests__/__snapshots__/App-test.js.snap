// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Listing components renders correctly Shallow rendering 1`] = `
<View
  style={
    Object {
      "height": 50,
      "margin": 5,
      "width": "92%",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "#d1d7db",
        "borderWidth": 1,
        "height": 40,
        "padding": 10,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;

exports[`Listing components renders correctly Shallow rendering 2`] = `
<View
  style={
    Object {
      "height": 50,
      "margin": 5,
      "width": "92%",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "#d1d7db",
        "borderWidth": 1,
        "height": 40,
        "padding": 10,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
